packages/core/src/editor/BlockNoteExtensions.ts

```diff
@@ -159,13 +165,45 @@ const getTipTapExtensions = <
     // marks:
     Link.extend({
       inclusive: false,
-      addKeyboardShortcuts() {
-        return {
-          "Mod-k": () => {
-            this.editor.commands.toggleLink({ href: "" });
-            return true;
-          },
-        };
+      // Adapted from https://github.com/ueberdosis/tiptap/blob/a0d2f2803652851bbe2f06f124a70bc01cfb0dab/packages/extension-link/src/link.ts#L301
+      // Fixes hrefs without a protocol prefix. Normally, these are treated as
+      // relative URLs, which doesn't make much sense for users. So if href is
+      // e.g. "www.google.com", it will now redirect users to:
+      // "https://www.google.com"
+      // instead of
+      // "<protocol>://<current-domain>/www.google.com".
+      renderHTML({ HTMLAttributes }) {
+        // prevent XSS attacks
+        if (!isAllowedUri(HTMLAttributes.href, this.options.protocols)) {
+          // strip out the href
+          return [
+            "a",
+            mergeAttributes(this.options.HTMLAttributes, {
+              ...HTMLAttributes,
+              href: "",
+            }),
+            0,
+          ];
+        }
+
+        // Modified section
+        // If the href does not start with a protocol, prefix with default one.
+        let prefix = this.options.defaultProtocol + "://";
+        for (const protocol of this.options.protocols) {
```

Comment:
lets use `https` as defaultProtocol (I think valid choice in 2024). We can do this where we add the extension